cmake/machinecfg/RZAnsel.cmake

```diff
@@ -0,0 +1,246 @@
+
+# It would be nice if we could let this variable float with the current code
+# checkout, but unfortunately CMake caches enough other stuff (like find
+# packages), that it's easier to just issue a warning if the view date is
+# different from the "latest" date.
+set(RZANSEL_VIEW_DATE ${RZANSEL_VIEW_DATE_LATEST}
+  CACHE STRING "RZAnsel dependency view being used")
+if (NOT RZANSEL_VIEW_DATE_LATEST STREQUAL RZANSEL_VIEW_DATE)
+    message(WARNING "Your current build directory was configured with a \
+    set of RZAnasel dependencies from ${RZANSEL_VIEW_DATE}, but your current \
+    code checkout prefers a set of RZAnsel dependencies from \
+        ${RZANSEL_VIEW_DATE_LATEST}. Consider configuring a new build \
+        directory.")
+endif()
+
+if (NOT DEFINED RZANSEL_COMPILER)
+    set(RZANSEL_COMPILER ${RZANSEL_COMPILER_PREFERRED})
+endif()
+
+if (RZANSEL_COMPILER STREQUAL "GCC")
+    set(RZANSEL_COMPILER ${RZANSEL_GCC_PREFERRED})
+endif()
+
+set(RZANSEL_PROJECT_PREFIX /usr/gapps/parthenon_shared/parthenon-project
+    CACHE STRING "Path to parthenon-project checkout")
+
+mark_as_advanced(RZANSEL_PROJECT_PREFIX)
+
+message(STATUS "RZAansel Build Settings
+              RZANSEL_ARCH: ${RZANSEL_ARCH}
+         RZANSEL_VIEW_DATE: ${RZANSEL_VIEW_DATE}
+          RZANSEL_COMPILER: ${RZANSEL_COMPILER}
+              RZANSEL_CUDA: ${RZANSEL_CUDA}
+    RZANSEL_PROJECT_PREFIX: ${RZANSEL_PROJECT_PREFIX}
+                 GPU_COUNT: ${GPU_COUNT}
+           MACHINE_VARIANT: ${MACHINE_VARIANT}
+")
+
+set(RZANSEL_ARCH_PREFIX ${RZANSEL_PROJECT_PREFIX}/views/rzansel/${RZANSEL_ARCH})
+
+if (NOT EXISTS ${RZANSEL_ARCH_PREFIX})
+    message(WARNING "No dependencies detected for \
+        RZANSEL_ARCH=\"${RZANSEL_ARCH}\" at ${RZANSEL_ARCH_PREFIX}")
+    return()
+endif()
+
+string(TOLOWER "${RZANSEL_COMPILER}" RZANSEL_COMPILER_LOWER)
+set(RZANSEL_COMPILER_PREFIX ${RZANSEL_ARCH_PREFIX}/${RZANSEL_COMPILER_LOWER})
+
+if (NOT EXISTS ${RZANSEL_COMPILER_PREFIX})
+    message(WARNING "No dependencies detected for \
+        RZANSEL_COMPILER=\"${RZANSEL_COMPILER}\" at ${RZANSEL_COMPILER_PREFIX}")
+    return()
+endif()
+
+set(RZANSEL_VIEW_PREFIX ${RZANSEL_COMPILER_PREFIX}${RZANSEL_VARIANT_PREFIX}/${RZANSEL_VIEW_DATE})
+if (NOT EXISTS ${RZANSEL_VIEW_PREFIX})
+    message(WARNING "No view detected for \
+        RZANSEL_VIEW_DATE=\"${RZANSEL_VIEW_DATE}\" at ${RZANSEL_VIEW_PREFIX}")
+    return()
+endif()
+
+if (RZANSEL_CUDA)
+    # Location of CUDA
+    set(CUDAToolkit_ROOT /usr/tce/packages/cuda/cuda-${RZANSEL_CUDA_VERSION} 
+      CACHE STRING "CUDA Location")
+
+    # All of this code ensures that the CUDA build uses the correct nvcc, and
+    # that we don't have to depend on the user's environment. As of this
+    # writing, nvcc_wrapper can only find nvcc by way of the PATH variable -
+    # there's no way to specify it in the command line. Therefore, we need to
+    # make sure that the expected nvcc is in the PATH when nvcc_wrapper is
+    # executed.
```

Comment:
I think nvcc_wraper can use the `CUDA_ROOT` environment variable to find nvcc:
```bash
# Default to whatever is in the path
nvcc_compiler=nvcc
if [ ! -z $CUDA_ROOT ]; then
  nvcc_compiler="$CUDA_ROOT/bin/nvcc"
fi
```

Reply:
@AndrewGaspar do you want to comment this section was basically copied from Darwin machine config.

Reply:
Oh, sure, this change was added in Kokkos 3.2.00, and I think when I originally wrote this code, we weren't using Kokkos 3.2.00. Using `CUDA_ROOT` is a better solution.